### PR TITLE
Mimikatz: Fix duplication of mimikatz credentials

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3168 .

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [ ] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py [hunk 1]
```diff
@@ -35,9 +35,10 @@ def __init__(
     def run(self, *, options=None, interrupt=None) -> Sequence[Credentials]:
         logger.info("Attempting to collect windows credentials with pypykatz.")
         windows_credentials = get_windows_creds()
-        logger.info(f"Pypykatz gathered {len(windows_credentials)} credentials.")
+        unique_credentials = list(set(windows_credentials))
+        logger.info(f"Pypykatz gathered {len(unique_credentials)} credentials.")
```
Comment (shreyamalviya) [inline monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py:39]:
```suggestion
        logger.info(f"Pypykatz gathered {len(unique_credentials)} unique credentials.")
```

File: monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py [hunk 2]
```diff
@@ -35,9 +35,10 @@ def __init__(
...
 
-        collected_credentials = self._to_credentials(windows_credentials)
+        collected_credentials = self._to_credentials(unique_credentials)
```
Comment (mssalvatore) [inline monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py:41]: This is fine. The other option is

```
collected_credentials = list(set(self._to_credentials(windows_credentials)))
logger.info(f"Pypykatz gathered {len(collected_credentials)} credentials.")
```

Then you don't meed to modify `WindowsCredentials` at all.

I'll leave the choice to you.

**Review by mssalvatore — APPROVED** [review]

File: monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py:41]: That seems to not work for certain test:
image

Comment (shreyamalviya) [reply monkey/agent_plugins/credentials_collectors/mimikatz/src/plugin.py]: Switch the order of credentials in `expected_credentials` in the test and try.

**Pull request merged**